src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -39,6 +39,45 @@ import java.io.IOException;
         }
     {% endfor %}
 
+    public static class {{ model.name }}Builder {
+        {% for field_name, field_val in struct_info["fields"] -%}
+            {% set field_value = field_val.0 | fully_qualified_type_name %}
+            private java.util.Optional<{{ field_value | wrapper_class }}> {{ field_name | camel }};
+        {% endfor %}
+
+        {{ model.name }}Builder() {
+            {# Initializes all the fields of this class #}
+            {% for field_name, field_val in struct_info["fields"] -%}
+                {% set field_value = field_val.0 | fully_qualified_type_name %}
+                {% set field_occurrence = field_val.1 %}
+                {# Initialize all the required fields as `java.util.Optional` to verify if the required fields were missing or not #}
+                this.{{ field_name | camel }} = java.util.Optional.empty();
+            {% endfor %}
+        }
+
+        {% for field_name, field_val in struct_info["fields"] %}
+            {% set field_value = field_val.0 %}
+            {% set val = field_value | fully_qualified_type_name %}
+            public {{ model.name }}Builder {{ field_name | camel }}({{ val }} {{ field_name | camel }}) {
+                this.{{ field_name | camel }} = java.util.Optional.of({{ field_name | camel }});
```

Comment:
Since this takes (e.g.) an `Integer`, we have to check for `null`.

```suggestion
                this.{{ field_name | camel }} = java.util.Optional.ofNullable({{ field_name | camel }});
```